test/unit_test/options_test.cc

```diff
@@ -83,3 +83,106 @@ BOOST_AUTO_TEST_CASE(create_argument_group) {
   BOOST_CHECK_EQUAL(ag.m_options[3]->m_keep, true);
   BOOST_CHECK_EQUAL(ag.m_options[3]->m_type_hash, typeid(decltype(loc2)).hash_code());
 }
+
+BOOST_AUTO_TEST_CASE(name_extraction_from_option_group)
+{
+  char loc;
+  std::vector<std::string> loc2;
+  option_group_definition ag("g1");
+  ag(make_option("im_necessary", loc).keep().necessary());
+  ag(make_option("opt2", loc));
+  ag.add(make_option("opt3", loc2));
+  ag.add(make_option("opt4", loc2).keep());
+
+  auto name_extractor = options_name_extractor();
+  // result should always be false
+  auto result = name_extractor.add_parse_and_check_necessary(ag);
```

Comment:
Nit: usage of auto

Reply:
I'll update this to bool to be more clear

Reply:
done